### PR TITLE
feat: add chips

### LINKED ISSUE #64: feat: add chip component
**Decathlon.design link**

https://www.decathlon.design/726f8c765/p/08dbe1-chip/b/10cbe3

**Material.io link**

https://material.io/components/chips

### PR DESCRIPTION
**Pull request by MaximeJineMaraval**
## Changes description

feat: Add chips component

## Context

close #64 

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on a phone device/emulator.
- [x] I have tested on a tablet device/emulator.
- [x] I have tested on a large screen device/emulator.
- [ ] If it includes design changes, please ask for a review with a core team designer.

## Does this introduce a breaking change?

- No

#### Phone

ezgif com-gif-maker

Screenshot_1645460655

#### Tablet

Screenshot_1645460220

#### Large screen

Screenshot_1645460145

Screenshot_1645460118

### CODE REVIEW

File: chips/README.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+    <com.decathlon.vitamin.chips.VitaminInputMediumChip
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:chipIcon="@drawable/ic_vtmn_map_pin_line"
+        android:text="Input medium with icon" />
+
+</com.google.android.material.chip.ChipGroup>
+```
+
+If you want to use image (like profile picture), you have a specific type for that (to avoid blue tint). You have to round the image by yourself.
```
Comment (GerardPaligot) [inline chips/README.md:115]: That could be great to show how to get a bitmap from Glide, convert it to a drawable and initialize a chip with this drawable.

File: chips/README.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+```xml
+<com.google.android.material.chip.ChipGroup
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <com.decathlon.vitamin.chips.VitaminInputImageMediumChip
```
Comment (GerardPaligot) [inline chips/README.md:122]: `Input`? That it means we can't have a drawable if it is another type? (`Filter`, `SingleChoice`, ...)

Comment (GerardPaligot) [reply chips/README.md:122]: If it is available for all types, that could be interesting to have a global section for all customizations available and not included in a specific type?

File: chips/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    <dimen name="vtmn_chip_min_height_medium">40dp</dimen>
+    <dimen name="vtmn_chip_min_height_small">32dp</dimen>
+    <dimen name="vtmn_chip_stroke_width">1dp</dimen>
+    <dimen name="vtmn_chip_text_start_padding_medium">16dp</dimen>
+    <dimen name="vtmn_chip_text_start_padding_small">12dp</dimen>
+    <dimen name="vtmn_chip_text_end_padding_medium">16dp</dimen>
+    <dimen name="vtmn_chip_text_end_padding_small">12dp</dimen>
+    <dimen name="vtmn_chip_icon_size">20dp</dimen>
+    <dimen name="vtmn_chip_w_start_ic_chip_start_padding_medium">8dp</dimen>
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply chips/src/main/res/values/dimens.xml:12]: +1, without the comment of @charlottemougenot, impossible to know what is "w"

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

File: chips/README.md (hunk 2 above)
Comment (MaximeJineMaraval) [reply chips/README.md:122]: It's only available for the input in the Vitamin design spec, but I will check that with a designer

File: chips/README.md (hunk 1 above)
Comment (MaximeJineMaraval) [reply chips/README.md:115]: Good idea, I will try that and put the expected code here

**Review by GerardPaligot — APPROVED** [review]

**Pull request merged**